src/OrchardCore/OrchardCore.Data.YesSql.Abstractions/DbConnectionValidatorResult.cs

```diff
@@ -2,21 +2,39 @@ namespace OrchardCore.Data;
 
 public enum DbConnectionValidatorResult
 {
-    // Unknown indicates that the connection string status is unknown or was not yet validated
+    /// <summary>
+    /// 'Unknown' indicates that the connection string status is unknown or was not validated.
+    /// </summary>
     Unknown,
 
-    // NoProvider indicated that the provider is missing
+    /// <summary>
+    /// 'NoProvider' indicated that the database provider is missing.
+    /// </summary>
     NoProvider,
 
-    // DocumentNotFound indicates that the connection string was valid, yet the Document table does not exist 
-    DocumentNotFound,
+    /// <summary>
+    /// 'DocumentTableNotFound' indicates that the connection string was valid, while the Document table does not exists.
+    /// </summary>
+    DocumentTableNotFound,
 
-    // DocumentFound indicates that the connection string was valid, yet the Document table exist
-    DocumentFound,
+    /// <summary>
+    /// 'DocumentTableFound' indicates that the connection string was valid, while the Document table exists.
+    /// </summary>
+    DocumentTableFound,
 
-    // InvalidConnection unable to open a connection to the given connection string
+    /// <summary>
+    /// 'ShellDescriptorDocumentFound' indicates that the connection string was valid and the document table exists with at least one shell-descriptor document.
+    /// This also means that the Document table is used by a tenant.
+    /// </summary>
+    ShellDescriptorDocumentFound,
+
+    /// <summary>
+    /// 'InvalidConnection' unable to open a connection to the given connection string.
+    /// </summary>
```

Comment:
I'm not sure why its named `Invalid` while the connection string is valid but unable to open!! May be we could rename it to `InUse`

Reply:
Sorry, I miss understood, I though you mean open the database not the connection ;)